Week-6/6.5/index.js

```diff
@@ -0,0 +1,61 @@
+function binarySearch(arr, start, end, target) {
+  let mid = Math.floor((start + end) / 2);
+
+  while (start <= end) {
+    if (arr[mid] === target) {
+      return true;
+    } else if (arr[mid] > target) {
+      end = mid - 1;
+    } else {
+      start = mid + 1;
+    }
+    mid = Math.floor((start + end) / 2);
+  }
+
+  return false;
+}
+
+// Time Complexity - O(nlogn)
+function findPairWithDifference(arr, target) {
```

Comment:
can you explain each of the strategy with simple language or add some comments in code so that I can be sure that you didn't blindly copied the code and understood the solution properly.

Reply:
Okay, I will do that in the next assignments.